sklearn/svm/_classes.py

```diff
@@ -1140,6 +1141,46 @@ def __init__(
             random_state=random_state,
         )
 
+    def fit(self, X, y, sample_weight=None):
+        """
+        Fit the model according to the given training data.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples
+            and n_features is the number of features.
+
+        y : array-like of shape (n_samples,)
+            Target values (class labels).
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Per-sample weights. If None, then each sample is given equal weight.
+
+        Returns
+        -------
+        self : object
+            Fitted estimator.
+        """
+        sample_weight = _check_sample_weight(sample_weight, X, dtype=np.float64)
+
+        if self.class_weight is not None:
+            # Compute class_weight_vect from class_weight
+            classes_ = np.unique(y)
+            class_weight_vect = compute_class_weight(
+                self.class_weight, classes=classes_, y=y
+            )
+
+            # Initialize sample_weight to 1 for all samples
+            sample_weight = np.ones(len(y), dtype=np.float64)
```

Comment:
The call to `_check_sample_weight` should already have done that if `sample_weight` is not. When not `None`, we should multiply the class weights with the user provided sample weight to combine the two.

```suggestion
```